### PR TITLE
grub2: update to 2.12

### PR DESCRIPTION
**Pull request by hackpascal**
New in 2.12:

* GCC 13 support.
* clang 14 support.
* binutils 2.38 support.
* Unification of EFI Linux kernel loader across architectures.
* Transition to EFI Linux kernel stub loader for x86 architecture.
* Initial support for Boot Loader Interface.
* Support for dynamic GRUB runtime memory addition using firmware calls.
* PCI and MMIO UARTs support.
* SDL2 support.
* LoongArch support.
* TPM driver fixes.
* Many filesystems fixes.
* Many CVE and Coverity fixes.
* Debugging support improvements.
* Tests improvements.
* Documentation improvements.
* ...and tons of other fixes and cleanups...

Tested platforms:
| Platform | Firmware | Boot device |
| --- | --- | --- |
| x86 | CSM | USB Drive |
| x86 | CSM | CD Drive |
| x86_64 | CSM | USB Drive |
| x86_64 | CSM | CD Drive |
| x86_64 | UEFI * | USB Drive |
| x86_64 | UEFI * | CD Drive |
| loongarch64 | UEFI * | USB Drive |

\* Secure Boot disabled

### CODE REVIEW

File: package/boot/grub2/Makefile [hunk 1]
```diff
@@ -6,12 +6,12 @@ include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=grub
-PKG_VERSION:=2.06
+PKG_VERSION:=2.12
 PKG_RELEASE:=6
```
Comment (PolynomialDivision) [inline package/boot/grub2/Makefile:10]: you can set PKG_RELEASE to 1

**Comment by hauke** [comment]
How did you test this on x86?

**Comment by hackpascal** [comment]
> How did you test this on x86?

@hauke I'v updated table of tested platforms. All tested on a desktop computer.

File: package/boot/grub2/Makefile (hunk 1 above)
Comment (hackpascal) [reply package/boot/grub2/Makefile]: done. Thx.

**Pull request merged**